### PR TITLE
Turn back on macOS shard Cirrus caching

### PR DESCRIPTION
**Pull request by jmagman**
## Description

This was previously attempted in https://github.com/flutter/flutter/pull/50496 and #51516 and reverted or closed due to longer cache miss times/network issues.

Maybe cache misses will be less frequent with `reupload_on_changes` off? https://github.com/flutter/flutter/pull/51524

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -547,7 +560,7 @@ task:
     - git fetch origin master # To set FETCH_HEAD, so that "git merge-base" works.
     - flutter config --no-analytics
     - flutter doctor -v
-    - flutter update-packages
+    - flutter update-packages --offline
```
Comment (christopherfujino) [inline .cirrus.yml:563]: what happens if this commit adds a new dependency/upgrades a version that isn't present locally?

Comment (jmagman) [reply .cirrus.yml:565]: The fingerprint checks the pubspec.yaml for their checksum, so it should be a cache miss and repopulate if something changes `echo $OS; grep -r --include=pubspec.yaml 'PUBSPEC CHECKSUM' "$CIRRUS_WORKING_DIR"`

Comment (christopherfujino) [reply .cirrus.yml:565]: Oh yeah...

**Comment by jmagman** [comment]
Nope.  All the mac tests took longer on a cache hit because the unzipping failed twice, then the cache had to repopulate anyway.
```
echo $OS; cat bin/internal/*.version
darwin
c332675a8c2f0c0ff32c2c5934089a8abb06e586
_d0dW3bRIm4FPvfx9G91NcQljDWxqqfgvnwJltaQDhoC
YGK_MICcFV1D95HW35kLaRRKRpoIIMWqqQTUcKJSFCQC
flutter_infra/gradle-wrapper/fd5c1f2c013565a3bea56ada6df9d2b8e96d56aa/gradle-wrapper.tgz
73b9c81e02f83060aa44bece963a62827908de68
2fbbee77ea2b3212959b9dddda816f59094cd7cd
6150fcd888a0ee8a69dd47328509ee608c2e02dc
28812e5bed8621bd637b4129ab15a9c1abb9f57d
8fb14f99a8854d4e2b16559c0eb48e7c297065ce
flutter_infra/flutter/fonts/13ac995daa9dda0a6ba0a45f1fccc541e616a74c/fonts.zip
2ef3f4c83bce209247479e707eebd5a84f606e28
eb45fd21469fa35e9bbc3b0a15a962460b8d23a2
Downloaded 34Mb.
Cache hit for all_flutter-63d900186d3a60312e13bfde6674e6c916430d346ccad2437a8b19d1c757898c!
Failed to unarchive all_flutter cache because of /tmp/flutter sdk/bin/cache/artifacts/engine/android-arm64-profile/darwin-x64/gen_snapshot: writing file after 157192 bytes (expected 10830404): unexpected EOF! Retrying...
Downloaded 50Mb.
Cache hit for all_flutter-63d900186d3a60312e13bfde6674e6c916430d346ccad2437a8b19d1c757898c!
Failed again to unarchive all_flutter cache because of /tmp/flutter sdk/bin/cache/artifacts/engine/android-x64/flutter.jar: writing file after 789011 bytes (expected 11100217): unexpected EOF!
Treating this failure as a cache miss! Cleaning up /tmp/flutter sdk/bin/cache...
Cache miss for all_flutter-63d900186d3a60312e13bfde6674e6c916430d346ccad2437a8b19d1c757898c! Populating...
git fetch origin
```

**Pull request closed without merging**